core/src/main/scala/quasar/datasource/kafka/SeekConsumer.scala

```diff
@@ -34,22 +34,34 @@ import scala.collection.immutable.SortedSet
  * A [[Consumer]] that fetches all messages available in a topic up to the last message
  * at the time it starts running.
  */
-class FullConsumer[F[_]: ConcurrentEffect: ContextShift: Timer, K, V](
+class SeekConsumer[F[_]: ConcurrentEffect: ContextShift: Timer, K, V](
+    initialOffsets: Offsets,
     settings: ConsumerSettings[F, K, V],
     decoder: RecordDecoder[F, K, V])
     extends Consumer[F] with Logging {
 
-  import FullConsumer.topicPartitionOrder
+  import SeekConsumer.topicPartitionOrder
 
   val F: ConcurrentEffect[F] = ConcurrentEffect[F]
 
-  override def fetch(topic: String): Resource[F, Stream[F, Byte]] = {
+  override def fetch(topic: String): Resource[F, (Offsets, Stream[F, Byte])] = {
     consumerResource[F].using(settings) evalMap { consumer =>
       for {
         endOffsets <- assignNonEmptyPartitionsForTopic(consumer, topic)
+        _ <- endOffsets.toList.traverse_ { case (p, end) =>
+          // `w - 1` because if `w` then in case if there is no new records
+          // CCR stream halts
+          initialOffsets.get(p.partition()).traverse_(w => consumer.seek(p, w - 1))
```

Comment:
Do we know what happens if we try and seek to a position < beginOffset or > endOffset? Seems like we might want to detect that and error if so? Being outside the partition range would seem to imply we're out of sync and probably missed some data.

We could have the strategy configurable so the user can decide if we should just pickup from the current beginning/end of the partition when out of range (depending on which bound we're outside of) or error.

Reply:
`beginOffset`s are `Map(whatever -> 0)` because we set seek from begining policy, but yeah, if `w < 0` this would be an error, which is impossible, due to saving offsets, and they're always greater than `0` (not even greater or equal, but greater). 

If I understand correctly `> endOffset` would be waiting for this offset to appear.